### Getting an idea of the wrong categorizations (manual)

In [1]:
import numpy as np
import pandas as pd

In [2]:
FILE_PATH = (
    "/Users/saranyapal/Developer/grid-determination-bizom/outputs/results_gmm.csv"
)
df = pd.read_csv(FILE_PATH)
# df.drop(column[0], axis=1, inplace=True)
df.shape

(2000, 23)

In [3]:
df.columns

Index(['Unnamed: 0', 'grid_id', 'latitude', 'longitude', 'building_count',
       'yellow_building_count', 'building_density', 'yellow_building_density',
       'total_building_density', 'main_road_density', 'local_road_density',
       'total_road_density', 'waterbody', 'water_percent', 'empty',
       'local_road_proportion', 'main_road_proportion',
       'yellow_building_proportion', 'other_building_proportion', 'labels',
       'prob_cluster_0', 'prob_cluster_1', 'prob_cluster_2'],
      dtype='str')

### Sorting by empty area percent (for identifying mis-classifications)

In [4]:
#   All wrong-classifications (From an empty% perspective)

ana_df = df[df["labels"] != 0].sort_values(by="empty", ascending=False)
new_ana_df = ana_df[
    [
        "grid_id",
        "labels",
        "empty",
        "building_density",
        "yellow_building_density",
        "total_building_density",
        "main_road_density",
        "local_road_density",
        "total_road_density",
        "water_percent",
        "local_road_proportion",
        "main_road_proportion",
        "yellow_building_proportion",
        "other_building_proportion",
    ]
].reset_index(drop=True)

new_ana_df[new_ana_df["empty"] > 90.0].shape

(127, 14)

In [ ]:
#   These are wrong classifications probably...
for _, row in new_ana_df[new_ana_df["empty"] > 90.0][["grid_id", "labels"]].iterrows():
    print(row["grid_id"], row["labels"])

36251 2
59442 1
45771 2
33050 1
63146 2
50531 1
22020 2
40546 1
29203 2
54962 1
18852 1
25402 2
12778 1
10984 1
2145 1
12922 1
47612 2
60366 1
5177 1
60312 1
59526 1
759 1
58949 1
12154 1
25615 1
63646 1
22132 1
31494 1
26626 1
54054 2
21658 1
51058 1
23397 1
63385 1
25448 1
10367 2
60862 1
26202 1
29056 1
62222 1
16405 1
58871 1
32520 1
23099 1
38486 1
41429 1
57291 2
31932 1
51816 1
58219 2
6444 1
62033 1
7203 1
57147 2
7296 1
39830 2
55560 1
18613 1
55722 1
59644 1
21905 1
30491 2
17441 1
26639 1
26883 1
61421 2
5635 1
18339 2
59099 1
20264 1
2883 1
4048 1
56556 1
21260 1
23869 1
49399 1
26874 1
53212 1
43008 2
12686 1
53213 1
13544 1
59798 1
7885 1
14260 1
52166 1
11483 2
20381 1
2868 1
46270 1
6596 1
19110 1
53244 1
2642 1
59429 1
5642 1
28524 1
32835 1
35226 1
26895 1
16510 1
16632 1
61581 2
15532 1
43582 1
4626 1
35704 1
21986 1
54194 1
40519 1
36917 1
2835 1
62117 1
16764 1
7303 1
11121 1
58463 1
5868 1
4436 1
24180 1
48712 1
4346 1
27326 1
43402 1
15791 1
27027 1
51798 1


### Sorting by water percent (for identifying mis-classifications)


In [6]:
df.sort_values(by="water_percent", ascending=False)

,Unnamed: 0,grid_id,latitude,longitude,building_count,yellow_building_count,building_density,yellow_building_density,total_building_density,main_road_density,...,water_percent,empty,local_road_proportion,main_road_proportion,yellow_building_proportion,other_building_proportion,labels,prob_cluster_0,prob_cluster_1,prob_cluster_2
48,48,38273,28.685195,77.231218,0,0,0.00,0.00,0.00,0.00,...,84.60,15.40,0.000000,0.000000,0.000000,0.000000,2,0.000000e+00,0.000000e+00,1.000000e+00
224,224,64211,28.881027,77.223030,0,0,0.00,0.00,0.00,0.00,...,64.60,35.40,0.000000,0.000000,0.000000,0.000000,2,0.000000e+00,0.000000e+00,1.000000e+00
245,245,45317,28.739094,77.034698,0,5,0.00,0.00,0.00,0.10,...,59.95,36.52,0.971671,0.028329,0.000000,0.000000,2,0.000000e+00,0.000000e+00,1.000000e+00
199,199,63970,28.879230,77.216889,0,0,0.00,0.00,0.00,0.00,...,57.16,42.84,0.000000,0.000000,0.000000,0.000000,2,0.000000e+00,0.000000e+00,1.000000e+00
324,324,25555,28.589974,77.018321,2,4,0.06,0.00,0.06,0.00,...,43.14,52.24,1.000000,0.000000,0.000000,1.000000,2,0.000000e+00,1.869636e-205,1.000000e+00
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
695,695,52155,28.791196,76.903684,0,0,0.00,0.00,0.00,0.00,...,0.00,100.00,0.000000,0.000000,0.000000,0.000000,0,1.000000e+00,0.000000e+00,3.994641e-09
694,694,42630,28.719331,76.893449,2,4,0.09,0.00,0.09,0.00,...,0.00,95.50,1.000000,0.000000,0.000000,1.000000,0,9.993440e-01,6.559551e-04,4.178882e-09
693,693,26639,28.597160,77.288537,24,7,1.14,0.00,1.14,0.00,...,0.00,93.87,1.000000,0.000000,0.000000,1.000000,1,2.261536e-84,9.999929e-01,7.135003e-06
692,692,7831,28.455227,77.276254,2,0,0.14,0.00,0.14,0.00,...,0.00,95.67,1.000000,0.000000,0.000000,1.000000,0,9.995460e-01,4.540053e-04,3.195509e-09


### This is a better approach than the above

In [ ]:
#   undeveloped_area = water_area + empty_area
df["undeveloped_percent"] = df["water_percent"] + df["empty"]

In [8]:
undeveloped_df = df[df["labels"] != 0].sort_values(
    by="undeveloped_percent", ascending=False
)
undeveloped_df[undeveloped_df["undeveloped_percent"] > 90.0].shape

(139, 24)

In [9]:
#   All of these should be labelled as 0
undeveloped_df[undeveloped_df["undeveloped_percent"] > 90.0].head()

,Unnamed: 0,grid_id,latitude,longitude,building_count,yellow_building_count,building_density,yellow_building_density,total_building_density,main_road_density,...,empty,local_road_proportion,main_road_proportion,yellow_building_proportion,other_building_proportion,labels,prob_cluster_0,prob_cluster_1,prob_cluster_2,undeveloped_percent
199,199,63970,28.879230,77.216889,0,0,0.0,0.0,0.0,0.0,...,42.84,0.0,0.0,0.0,0.0,2,0.0,0.0,1.0,100.0
599,599,61581,28.861264,77.198465,0,0,0.0,0.0,0.0,0.0,...,91.89,0.0,0.0,0.0,0.0,2,0.0,0.0,1.0,100.0
224,224,64211,28.881027,77.223030,0,0,0.0,0.0,0.0,0.0,...,35.40,0.0,0.0,0.0,0.0,2,0.0,0.0,1.0,100.0
45,45,11404,28.482177,77.282396,0,3,0.0,0.0,0.0,0.0,...,80.92,0.0,0.0,0.0,0.0,2,0.0,0.0,1.0,100.0
48,48,38273,28.685195,77.231218,0,0,0.0,0.0,0.0,0.0,...,15.40,0.0,0.0,0.0,0.0,2,0.0,0.0,1.0,100.0


In [10]:
undeveloped_df[undeveloped_df["undeveloped_percent"] > 90.0].tail()

,Unnamed: 0,grid_id,latitude,longitude,building_count,yellow_building_count,building_density,yellow_building_density,total_building_density,main_road_density,...,empty,local_road_proportion,main_road_proportion,yellow_building_proportion,other_building_proportion,labels,prob_cluster_0,prob_cluster_1,prob_cluster_2,undeveloped_percent
437,437,15791,28.516312,77.006039,3,2,3.69,0.0,3.69,0.05,...,90.19,0.991830,0.008170,0.000000,1.000000,1,0.0,9.999932e-01,6.836480e-06,90.19
669,669,9106,28.466007,76.963050,4,10,1.03,0.0,1.03,0.00,...,89.58,1.000000,0.000000,0.000000,1.000000,1,0.0,9.999994e-01,6.275048e-07,90.15
316,316,34153,28.654652,77.079734,3,0,2.53,0.0,2.53,2.80,...,55.41,0.618529,0.381471,0.000000,1.000000,2,0.0,3.285911e-126,1.000000e+00,90.13
639,639,27027,28.600754,77.108393,9,0,2.92,0.0,2.92,0.06,...,90.08,0.991429,0.008571,0.000000,1.000000,1,0.0,9.999958e-01,4.172131e-06,90.08
185,185,51798,28.787602,77.147288,4,3,5.41,0.6,6.01,0.00,...,90.02,1.000000,0.000000,0.099834,0.900166,1,0.0,1.000000e+00,1.311956e-76,90.02


### Checking by building_counts (yellow and total)

In [11]:
yw_builds = df.sort_values(by="yellow_building_count", ascending=False).reset_index(
    drop=True
)
yw_builds

,Unnamed: 0,grid_id,latitude,longitude,building_count,yellow_building_count,building_density,yellow_building_density,total_building_density,main_road_density,...,empty,local_road_proportion,main_road_proportion,yellow_building_proportion,other_building_proportion,labels,prob_cluster_0,prob_cluster_1,prob_cluster_2,undeveloped_percent
0,1798,29384,28.618720,77.061310,78,67,30.41,25.72,56.13,1.45,...,24.11,0.926619,0.073381,0.458222,0.541778,1,0.0,1.0,0.000000e+00,24.11
1,1887,34228,28.654652,77.233266,91,64,31.10,31.00,62.10,0.00,...,32.95,1.000000,0.000000,0.499195,0.500805,1,0.0,1.0,0.000000e+00,32.95
2,1631,41298,28.708551,77.089969,71,56,54.07,12.08,66.15,0.06,...,17.32,0.996370,0.003630,0.182615,0.817385,1,0.0,1.0,0.000000e+00,17.32
3,1558,34957,28.660042,77.263972,79,54,35.88,27.34,63.22,0.70,...,18.91,0.960828,0.039172,0.432458,0.567542,1,0.0,1.0,0.000000e+00,18.91
4,1873,30564,28.627703,77.040839,62,51,45.27,16.31,61.58,0.08,...,23.47,0.994649,0.005351,0.264859,0.735141,1,0.0,1.0,0.000000e+00,23.47
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1995,327,57056,28.827128,77.192324,1,0,0.04,0.00,0.04,0.00,...,99.96,0.000000,0.000000,0.000000,1.000000,0,1.0,0.0,2.327240e-08,99.96
1996,328,54374,28.807365,77.061310,25,0,20.23,0.00,20.23,2.66,...,70.60,0.709924,0.290076,0.000000,1.000000,1,0.0,1.0,1.045922e-12,70.60
1997,628,64491,28.882823,77.309008,0,0,0.00,0.00,0.00,0.00,...,96.48,1.000000,0.000000,0.000000,0.000000,0,1.0,0.0,7.268429e-09,96.48
1998,331,23029,28.570211,77.206654,27,0,13.67,0.00,13.67,17.41,...,60.67,0.321512,0.678488,0.000000,1.000000,1,0.0,1.0,2.947815e-26,60.67


### Observation:
1. Yellow building areas are not correctly identified (probably)...[Eg: 51798 etc]
2. So, probably also need to include yellow building count
3. The proportion metrics are adding noise to the data...